networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/DiscoveryNetworkTest.java

```diff
@@ -174,6 +179,45 @@ public void setForkInfo_futureForkScheduled() {
         .updateCustomENRField("eth2", SimpleOffsetSerializer.serialize(expectedEnrForkId));
   }
 
+  @Test
+  public void setForkInfoShouldAddPredicateToConnectionManager() {
+    final ForkInfo currentForkInfo = dataStructureUtil.randomForkInfo();
+    discoveryNetwork.setForkInfo(currentForkInfo, Optional.empty());
+
+    final EnrForkId expectedEnrForkId =
+        new EnrForkId(
+            currentForkInfo.getForkDigest(),
+            currentForkInfo.getFork().getCurrent_version(),
+            FAR_FUTURE_EPOCH);
+    Bytes encodedForkId = SimpleOffsetSerializer.serialize(expectedEnrForkId);
+    verify(discoveryService).updateCustomENRField("eth2", encodedForkId);
+    ArgumentCaptor<ConnectionManager.PeerPredicate> peerPredicateArgumentCaptor =
+        ArgumentCaptor.forClass(ConnectionManager.PeerPredicate.class);
+    verify(connectionManager).addPeerPredicate(peerPredicateArgumentCaptor.capture());
+
+    DiscoveryPeer peer1 = mock(DiscoveryPeer.class);
```

Comment:
It's usually better not to mock data classes like `DiscoveryPeer`.  Just create a real one passing in the values you need it to return from it's getters.

Reply:
Done.